chanbackup/backupfile_test.go

```diff
@@ -45,160 +45,217 @@ func assertFileDeleted(t *testing.T, filePath string) {
 
-		valid bool
+	testCases := []struct {
+		name          string
+		oldTempExists bool
 	}{
-		// Main file name is blank, should fail.
-		{
-			fileName: "",
-			valid:    false,
-		},
-
 		// Old temporary file still exists, should be removed. Only one
 		// file should remain.
 		{
-			fileName: filepath.Join(
-				tempTestDir, DefaultBackupFileName,
-			),
-			tempFileName: filepath.Join(
-				tempTestDir, DefaultTempBackupFileName,
-			),
+			name:          "remove old temp file",
 			oldTempExists: true,
-			valid:         true,
 		},
 
 		// Old temp doesn't exist, should swap out file, only a single
 		// file remains.
 		{
-			fileName: filepath.Join(
-				tempTestDir, DefaultBackupFileName,
-			),
-			tempFileName: filepath.Join(
-				tempTestDir, DefaultTempBackupFileName,
-			),
-			valid: true,
+			name:          "swap out file",
+			oldTempExists: false,
 		},
+	}
 
-		// Test with noBackupArchive set to true - should not create
-		// archive.
-		{
-			fileName: filepath.Join(
-				tempTestDir, DefaultBackupFileName,
-			),
-			tempFileName: filepath.Join(
-				tempTestDir, DefaultTempBackupFileName,
-			),
-			noBackupArchive: true,
-			valid:           true,
-		},
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tempTestDir := t.TempDir()
 
-		// Test with v set to false - should create
-		// archive.
-		{
-			fileName: filepath.Join(
+			fileName := filepath.Join(
 				tempTestDir, DefaultBackupFileName,
-			),
-			tempFileName: filepath.Join(
+			)
+			tempFileName := filepath.Join(
 				tempTestDir, DefaultTempBackupFileName,
-			),
-			noBackupArchive: false,
-			valid:           true,
-		},
-	}
-	for i, testCase := range testCases {
-		backupFile := NewMultiFile(
-			testCase.fileName, testCase.noBackupArchive,
-		)
+			)
 
-		// To start with, we'll make a random byte slice that'll pose
-		// as our packed multi backup.
-		newPackedMulti, err := makeFakePackedMulti()
-		if err != nil {
-			t.Fatalf("unable to make test backup: %v", err)
-		}
+			backupFile := NewMultiFile(fileName, false)
 
-		// If the old temporary file is meant to exist, then we'll
-		// create it now as an empty file.
-		if testCase.oldTempExists {
-			f, err := os.Create(testCase.tempFileName)
-			if err != nil {
-				t.Fatalf("unable to create temp file: %v", err)
-			}
-			require.NoError(t, f.Close())
+			// To start with, we'll make a random byte slice that'll
+			// pose as our packed multi backup.
+			newPackedMulti, err := makeFakePackedMulti()
+			require.NoError(t, err)
 
-			// TODO(roasbeef): mock out fs calls?
-		}
+			// If the old temporary file is meant to exist, then
+			// we'll create it now as an empty file.
+			if tc.oldTempExists {
+				f, err := os.Create(tempFileName)
+				require.NoError(t, err)
+				require.NoError(t, f.Close())
 
-		// With our backup created, we'll now attempt to swap out this
-		// backup, for the old one.
-		err = backupFile.UpdateAndSwap(PackedMulti(newPackedMulti))
```

Comment:
Because a file was left on the disk from the previous test case, this call will create an archive.